packages/renderer/src/PreferencesNavigation.spec.ts

```diff
@@ -0,0 +1,66 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom';
+import { beforeAll, test, expect } from 'vitest';
+import { render, screen } from '@testing-library/svelte';
+import PreferencesNavigation from './PreferencesNavigation.svelte';
+
+// fake the window.events object
+beforeAll(() => {
+  (window.events as unknown) = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    receive: (_channel: string, func: any) => {
+      func();
+    },
+  };
+});
+
+test('Test that Settings navigation bar has right role and aria-label', () => {
+  render(PreferencesNavigation, {
+    meta: {
+      url: '/',
+    },
+  });
+
+  const navigationBar = screen.getByRole('navigation', { name: 'PreferencesNavigation' });
+  expect(navigationBar).toBeVisible();
+  expect('PreferencesNavigation').toEqual(navigationBar.getAttribute('aria-label'));
```

Comment:
not sure it's needed as when we fetch by role at line 41 it requires to be correctly defined